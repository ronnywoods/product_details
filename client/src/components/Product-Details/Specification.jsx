import React from 'react';

const Specification = props => (
  <div className="details container">
    <div className="row detail-row">
      <div className="col-sm first-col">
        <ul>
          {props.details.slice(0, Math.floor(props.details.length / 2)).map((detail, i) => (
            <li key={i}>{detail}</li>
          ))}
        </ul>
      </div>
      <div className="col-sm sec-col">
        <ul>
          {props.details.slice(Math.floor(props.details.length / 2)).map((detail, i) => (
            <li key={i}>{detail}</li>
          ))}
        </ul>
      </div>
    </div>
  </div>
);

export default Specification;
