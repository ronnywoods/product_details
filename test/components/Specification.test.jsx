import React from 'react';
import { shallow } from 'enzyme';
import Specification from '../../client/src/components/Product-Details/Specification';

describe('Specification Component', () => {
  const sample = ['Lace closure', 'Seam-sealed stretch-mesh upper',
    'Arch type: Normal', 'supportive feel', 'imported', 'Product Code: B31231'];

  const component = shallow(<Specification details={sample} />);

  it('should display like snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should display 2 columns of details', () => {
    expect(component.find('div.col-sm').length).toBe(2);
    expect(component.find('.first-col')).toExist();
    expect(component.find('.sec-col')).toExist();
  });

  it('should have correct amount of details', () => {
    expect(component.find('li').length).toBe(6);
  });

  it('should display details properly', () => {
    expect(component.find('li').at(0).text()).toBe('Lace closure');
    expect(component.find('li').at(2).text()).toBe('Arch type: Normal');
  });
});
