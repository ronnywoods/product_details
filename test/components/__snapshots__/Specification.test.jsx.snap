// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Specification Component should display like snapshot 1`] = `
<div
  className="details container"
>
  <div
    className="row detail-row"
  >
    <div
      className="col-sm first-col"
    >
      <ul>
        <li
          key="0"
        >
          Lace closure
        </li>
        <li
          key="1"
        >
          Seam-sealed stretch-mesh upper
        </li>
        <li
          key="2"
        >
          Arch type: Normal
        </li>
      </ul>
    </div>
    <div
      className="col-sm sec-col"
    >
      <ul>
        <li
          key="0"
        >
          supportive feel
        </li>
        <li
          key="1"
        >
          imported
        </li>
        <li
          key="2"
        >
          Product Code: B31231
        </li>
      </ul>
    </div>
  </div>
</div>
`;
